### PR TITLE
timeutil: introduce helpers for weekdays and TimeOfDay

### PR DESCRIPTION
**Pull request by bboozzoo**
Putting helpers introduced in  of #4269 in a separate PR

### CODE REVIEW

File: timeutil/schedule.go [hunk 1]
```diff
@@ -39,6 +39,30 @@ func (t TimeOfDay) String() string {
...
+	t2 := time.Duration(other.Hour)*time.Hour + time.Duration(other.Minute)*time.Minute
+	return t1 - t2
+}
+
+// Add adds given duration and returns a new TimeOfDay
+func (t TimeOfDay) Add(dur time.Duration) TimeOfDay {
+	t1 := time.Duration(t.Hour)*time.Hour + time.Duration(t.Minute)*time.Minute
+	t2 := t1 + dur
+	nt := TimeOfDay{
+		Hour:   int(t2.Hours()) % 24,
```
Comment (zyga) [inline timeutil/schedule.go]: Is this accurate for stuff like leap year and 60ths second? I would prefer if we could work at a level of simple seconds and convert to/from other representations.

File: timeutil/schedule.go [hunk 2]
```diff
@@ -39,6 +39,30 @@ func (t TimeOfDay) String() string {
...
+		Minute: int(t2.Minutes()) % 60,
+	}
+	return nt
+}
+
+// isValidWeekday returns true if given s looks like a valid weekday
+func isValidWeekday(s string) bool {
```
Comment (zyga) [inline timeutil/schedule.go]: Nitpick: this is not tested.

Comment (bboozzoo) [reply timeutil/schedule.go]: Added a test with invalid weekday.

File: timeutil/schedule.go (hunk 1 above)
Comment (bboozzoo) [reply timeutil/schedule.go]: Thanks for the commen. There's no date and location context so it's not possible for this to be aware of leap years, leap seconds (`time.Time()` is not aware of leap seconds either) or DST. It's using time.Duration() under the hood (which is just seconds), the calculations are done using `time.Hour` and `time.Minute` both are convertible to integer types without loss. 

IMO this should be fine for the purpose of just adding some duration to `TimeOfDay` and wrapping around.

File: timeutil/schedule.go (hunk 2 above)
Comment (zyga) [reply timeutil/schedule.go]: Thanks!

**Pull request merged**